six/modules/c++/six/include/six/ReadControl.h

```diff
@@ -108,12 +109,43 @@ class ReadControl
      *  This function reads in the image area specified by the region.
      *  If you want us to use a work buffer, you can set it through region.
      *
+     *  To simply read the entire image, pass a default-constructed
+     *  Region as the first parameter
+     *
      *  Once read, the image buffer is set in both the region pointer,
      *  and in the return value, for convenience
+     *
+     *  For safety, prefer the overload below.
      */
     virtual UByte* interleaved(Region& region, size_t imageNumber) = 0;
 
     /*!
+     *  This function reads in the image area specified by the region.
+     *  If you want us to use a work buffer, you can set it through region.
+     *
+     *  To simply read the entire image, pass a default-constructed
+     *  Region as the first parameter.
+     *
+     *  Once read, the image buffer is set in the passed buffer, and
+     *  also the region pointer and the return value for convenience.
+     *
+     * \param region Region stores row and cols to be read
+     * \param imageNumber Index of the image to read
+     * \param buffer Scoped array that holds the memory for the read-in image.
+     * This will be allocated by this function.
+     *
+     * \return Buffer of image data.  This is simply equal to buffer.get() and
+     * is provided as a convenience.
+     */
+    template<typename T>
+    UByte* interleaved(Region& region, size_t imageNumber,
+            mem::ScopedArray<T>& buffer)
+    {
+        buffer.reset(reinterpret_cast<T*>(interleaved(region, imageNumber)));
```

Comment:
So... if they pass in a region with a non-NULL buffer, they will probably think they still own the memory and double delete it.  I wish the region held a smart pointer but it doesn't right now.  Any thoughs?